src/components/budget/handlers/update-project-current-budget-status.handler.ts

```diff
@@ -0,0 +1,48 @@
+import { EventsHandler, IEventHandler } from '../../../core';
+import { ProjectStatus } from '../../project';
+import { ProjectUpdatedEvent } from '../../project/events';
+import { BudgetService } from '../budget.service';
+import { BudgetStatus, UpdateBudget } from '../dto';
+
+@EventsHandler(ProjectUpdatedEvent)
+export class UpdateProjectBudgetStatusHandler
+  implements IEventHandler<ProjectUpdatedEvent> {
+  constructor(private readonly budgets: BudgetService) {}
+
+  private async updateBudgetStatusToCurrent({
```

Comment:
Can you just merge this with the handle method? Or split out this out into helper methods. Having handle() just forward to another method isn't useful.